### PR TITLE
Top helpers not askers

### LINKED ISSUE #762: Exclude OP (askers) from TopHelpersMessageListener
# About
Should be a good idea to not add messages to the database if its the asker that writes in his question.
Implementation would just be an if statement checking if a user is OP in a forum post.

### PR DESCRIPTION
**Pull request by Taz03**
closes #762

### CODE REVIEW

File: application/src/main/java/org/togetherjava/tjbot/features/tophelper/TopHelpersMessageListener.java [hunk 1]
```diff
@@ -83,6 +83,14 @@ boolean isHelpThread(MessageChannelUnion channel) {
         return isHelpForumName.test(rootChannelName);
     }
 
+    boolean isSentByOp(MessageReceivedEvent event) {
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/features/tophelper/TopHelpersMessageListener.java:86]: dont think its meaningful to expose this for now. id go for private

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Comment by Zabuzard** [comment]
TopHelper unit tests crash, because you introduced new methods into the flow that havent been mocked yet. Do you want to fix this urself or should someone else take that part over?

**Comment by Zabuzard** [comment]
Has to be done in the `JdaTestHelper` main class or whatever its called

**Comment by Taz03** [comment]
> TopHelper unit tests crash, because you introduced new methods into the flow that havent been mocked yet. Do you want to fix this urself or should someone else take that part over?

i haven't tried yet, i'll like to try that myself first, will update on it

**Comment by Taz03** [comment]
@Zabuzard for some reason when i use shorthand method it passes the test, is it ok?

**Comment by Zabuzard** [comment]
> 

i suppose that method is mocked already - lol!

fair enough, i guess

**Review by Zabuzard — APPROVED** [review]

**Review by SquidXTV — APPROVED** [review]

**Pull request merged**